src/cli-options.ts

```diff
@@ -17,14 +17,25 @@ import {QuietCiLogger, QuietLogger} from './logging/quiet-logger.js';
 import {DefaultLogger} from './logging/default-logger.js';
 
 export const packageDir = await (async (): Promise<string | undefined> => {
-  // Recent versions of npm set this environment variable that tells us the
-  // package.
-  const packageJsonPath = process.env.npm_package_json;
-  if (packageJsonPath) {
-    return pathlib.dirname(packageJsonPath);
+  // Recent versions of npm and yarn set the `npm_package_json` environment
+  // variable.
+  //
+  // yarn sets it to the package.json at the project root, even if we're in
+  // another package.
```

Comment:
To be clear, that's only when `-T` is specified. To quote the documentation, `-T` "checks the root workspace for scripts and/or binaries instead of the current one", hence why `npm_package_json` becomes the root one.

Reply:
If I recall correctly, the `-T` got added because `yarn run wireit` wasn't finding the root `wireit` when in a workspace.

So from the POV of `wireit`, it will always been the root package (since `wireit` is meant to be a `devDependency` of the whole project).

Reply:
If I recall correctly, the `-T` got added because `yarn run wireit` wasn't finding the root `wireit` when in a workspace.

So from the POV of `wireit`, it will always been the root package (since `wireit` is meant to be a `devDependency` of the whole project).

Reply:
Right, I just don't want someone external reading this comment to think "Wait what; Yarn is broken if it doesn't run `run` properly" or something of the kind 🙂